homeassistant/components/ecobee/weather.py

```diff
@@ -113,7 +114,10 @@ def temperature_unit(self):
     def pressure(self):
         """Return the pressure."""
         try:
-            return int(self.get_forecast(0, "pressure"))
+            pressure = self.get_forecast(0, "pressure")
+            if not self.hass.config.units.is_metric:
+                return pressure_convert(pressure, PRESSURE_HPA, PRESSURE_INHG)
```

Comment:
I would recommend rounding here too. Probably to two decimal places for InHG.

Reply:
Fixed!

Reply:
Looks good. Thanks!